src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -129,6 +129,15 @@
       <SourcesToCopyToTestDir Include="@(OverridenRuntimeContent)" />
       <DestinationsForTestDir Include="@(OverridenRuntimeContent->'$(TestPath)\%(Filename)%(Extension)')" />
     </ItemGroup>
+    
+    <!-- Removing assets from TestCopyLocal that come in from P2P references in SourcesToCopyToTestDir -->
+    <ItemGroup>
+        <_FileNamesInSourcesToCopyToTestDir Include="@(SourcesToCopyToTestDir->'%(Filename)%(Extension)')">
+          <SourcePath>%(SourcesToCopyToTestDir.Identity)</SourcePath>
+        </_FileNamesInSourcesToCopyToTestDir>
+        <_DeduppedListOfTestAssetsByFileName Include="@(_TestCopyLocalByFileNameWithoutDuplicates)" Exclude="@(_FileNamesInSourcesToCopyToTestDir)" />
+        <_DeduppedListOfTestAssetsByFileName Include="@(_FileNamesInSourcesToCopyToTestDir)" />
```

Comment:
/cc @MattGal now that we have a correct merged list of files we should look at removing the hacks from the test script and the VS target that only copy over if the file doesn't already exist.